include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -233,6 +234,16 @@ struct __atomic_ref : sycl::atomic<_AtomicType, _Space>
 };
 #endif // _ONEDPL_SYCL2023_ATOMIC_REF_PRESENT
 
+template <typename DataT, int Dimensions = 1>
+using __local_accessor =
+#if _ONEDPL_LIBSYCL_VERSION >= 60000
+    sycl::local_accessor<DataT, Dimensions>;
+#else
+    sycl::accessor<DataT, Dimensions,
+                   ::std::is_const_v<DataT> ? sycl::access::mode::read : sycl::access::mode::read_write,
```

Comment:
Local accessor does not have `read` mode.

According to [SYCL 2020](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_accessor_specialization_with_targetlocal):
> The accessor class may be specialized with target target::local ... This specialization of accessor is only available for access modes access_mode::read_write and access_mode::atomic.

Let's leave only `read_write` (we do not use `atomic` mode for now).

Reply:
thank Dmitriy that found that clarification... because I mentioned and raised a question what means read mode for SLM in one of previous comments.

Reply:
Thank you, @dmitriy-sobolev, fixed!

Reply:
@MikeDvorskiy, the quotation above is only about the deprecated `accessor<... target::local>` API. It is still unclear, at least for me, how to work with the read-only local accessor available in SYCL 2020 API.